src/posts/elastic-cloud-kubernetes-how-to-fix-a-corrupt-index-exception.md

```diff
@@ -0,0 +1,77 @@
+---
+type: Blog
+title: 'Elastic Cloud Kubernetes: How to fix a corrupt index exception'
+subheading: In this article we’re going to see how to fix a corrupt index which was not automatically recovered.
+authors:
+- avijayan
+tags:
+- Elastic
+date: 2022-11-18
+image: "/blogs/elastic-cloud-kubernetes-how-to-fix-a-corrupt-index-exception.png"
+featured: true
+
+---
+
+This article assumes the reader has good knowledge about the Elastic Stack. Also, this article tries to discuss a very specific problem happening in Elastic Stack environments.
+
+## Introduction
+
+To solve corruption problems, Elasticsearch database ships with a binary `elasticsearch-shard` which can remove the corrupt parts in a shard copy of a Lucene index or Translog. This tool comes in handy when the corrupt shard cannot be recovered automatically or restored from a backup. This can cause data loss. From the official documentation, *this approach should be chosen as the last resort when restoring from backups is not an option*. Also, sometimes, it’s hard to restore system indices from Elasticsearch’s snapshot backup.
+
+In order to execute this command in an Elasticsearch node, the Elasticsearch service should not be running. This is to prevent any read/write actions while the shard is being repaired. This is complicated in a container-based environment like Kubernetes because the container exits if the entry point process is stopped, in this case, the Elasticsearch process.
+
+However, Elastic Cloud Kubernetes comes with a way to suspend containers with the help of Kubernetes annotations. When annotated with [`eck.k8s.elastic.co/suspend`](http://eck.k8s.elastic.co/suspend) the container restarts into one of init containers, namely `elastic-internal-suspend` with all the necessary disks mounted. After connecting to that internal container, one can have access to the disk and can run commands.  
```

Comment:
Another reference to "Elastic Cloud Kubernetes". Should we add the "on" inbetween?

Reply:
Yes, please